lib/src/git.rs

```diff
@@ -110,6 +115,37 @@ fn to_remote_branch<'a>(parsed_ref: &'a RefName, remote_name: &str) -> Option<&'
     }
 }
 
+// there are two options:
+//
+// a bare git repo, which has a parent named .jj that sits on the workspace root
+// or
+// a colocated .git dir, which is already on the workspace root
```

Comment:
suggestion (non-blocking): I think it's probably better to promote this to a doc-comment (`///`)? (Assuming that this function makes it into the final version.)